cmake/modules/Findcrypto.cmake

```diff
@@ -0,0 +1,102 @@
+# - Try to find LibCrypto include dirs and libraries
+#
+# Usage of this module as follows:
+#
+#     find_package(crypto)
+#
+# Variables used by this module, they can change the default behaviour and need
+# to be set before calling find_package:
+#
+# Variables defined by this module:
+#
+#  Crypto_FOUND             System has libcrypto, include and library dirs found
```

Comment:
Let's make the capitalization consistent. Currently it's `crypto` in some places and `Crypto` in others:

Looking at other [cmake find scripts](https://cmake.org/cmake/help/latest/manual/cmake-modules.7.html#find-modules) I see there's not perfect consistency, but `CamelCase` seems most common, followed by `ALLCAPS`. `alllower` is not very common, but if it makes us consistent with other libs we're putting in the `AWS::` namespace then that's fine

but in any case let's pick one style and stick with it

Reply:
it's not entirely that simple. A lot of this is due to cmake differences across platforms and case-sensitivity. I'll see if i can get the build working with the pure lower case values, but I might just set both of them to be safe.